### PR TITLE
Task/drawer

### PR DESCRIPTION
**Pull request by azizalshrafi**
Add drawer to project

### CODE REVIEW

File: lib/core/constants/sizes_app.dart [hunk 1]
```diff
@@ -1,10 +1,26 @@
...
+  static final double w54 = 54.w;
+  static final double w11 = 11.39.w;
+  static final double w24 = 24.w;
+  static final double w23 = 23.w;
+  static final double w72 = 72.59.w;
 
-}
\ No newline at end of file
+  static final double h51 = 51.h;
+  static final double h20 = 20.2.h;
```
Comment (AbdalazizAbdallah) [inline lib/core/constants/sizes_app.dart:22]: remove point number

File: lib/core/constants/sizes_app.dart [hunk 2]
```diff
@@ -1,10 +1,26 @@
...
+  static final double r64 = 64.r;
+
+  static final double r150 = 150.r;
+  static final double r140 = 140.r;
+
+  static final double sp14 = 14.sp;
+  static final double sp12 = 12.sp;
+  static final double sp16 = 16.sp;
+
+  static final double w54 = 54.w;
```
Comment (AbdalazizAbdallah) [inline lib/core/constants/sizes_app.dart:15]: use r instead of w

File: lib/core/routes/routes_app.dart [hunk 3]
```diff
@@ -1,18 +1,27 @@
...
+      GetPage(
+        name: initialPage,
+        page: () => const HomePage(),
+      ),
+
+      GetPage(
+    name: drawer,
+    page: () => const MyDrawer(),
+    binding: DrawerBinding(),
+    fullscreenDialog: true,
```
Comment (AbdalazizAbdallah) [inline lib/core/routes/routes_app.dart:23]: useless here, we won't show a dialog

File: lib/home_page.dart [hunk 4]
```diff
@@ -1,15 +1,33 @@
 import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
 import 'package:get/get.dart';
+import 'package:gtc_customer/presentation/pages/drawer.dart';
+import 'core/constants/images_app.dart';
+import 'core/constants/sizes_app.dart';
+
+final GlobalKey<ScaffoldState> _key = GlobalKey();
```
Comment (AbdalazizAbdallah) [inline lib/home_page.dart:8]: add any Global Keys in controllers not inside page , to show your code WOW ^_^

File: lib/home_page.dart [hunk 5]
```diff
@@ -1,15 +1,33 @@
...
     return Scaffold(
+      key: _key,
       appBar: AppBar(
+        leading: IconButton(
+          onPressed: () {
+            _key.currentState!.openDrawer();
+          },
+          icon: SvgPicture.asset(
+            ImagesApp.menubar,
+            width: SizesApp.w23,
```
Comment (AbdalazizAbdallah) [inline lib/home_page.dart:24]: r instead of w, Is this square? so use same variable

File: lib/presentation/pages/drawer.dart [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        padding: EdgeInsets.only(top: 40),
+        children: [
+          
+          IconButton(
+            onPressed: () {
+              Get.back();
+            },
+            icon: SvgPicture.asset(
+              ImagesApp.back,
+              width: SizesApp.w24,
```
Comment (AbdalazizAbdallah) [inline lib/presentation/pages/drawer.dart:30]: same pervious review ^_^

File: lib/presentation/pages/drawer.dart [hunk 7]
```diff
@@ -0,0 +1,133 @@
...
+import 'package:gtc_customer/presentation/widget/custom_ListTile.dart';
+
+class MyDrawer extends GetView<DrawerController> {
+  const MyDrawer({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Drawer(
+      child: ListView(
+        padding: EdgeInsets.only(top: 40),
```
Comment (AbdalazizAbdallah) [inline lib/presentation/pages/drawer.dart:21]: put 40.r in SizesApp

File: lib/presentation/pages/drawer.dart [hunk 8]
```diff
@@ -0,0 +1,133 @@
...
+            icon: SvgPicture.asset(
+              ImagesApp.back,
+              width: SizesApp.w24,
+              height: SizesApp.h24,
+            ),
+            alignment: Alignment.centerLeft,
+          ),
+          Column(
+            children: [
+              Container(
```
Comment (AbdalazizAbdallah) [inline lib/presentation/pages/drawer.dart:37]: learn about DrawerHeader widgets and maybe use it

File: lib/presentation/pages/drawer.dart [hunk 9]
```diff
@@ -0,0 +1,133 @@
...
+                'email'.tr,
+                style: GoogleFonts.openSans(
+                  fontSize: SizesApp.sp12,
+                  color: ColorsApp.grey,
+                ),
+              ),
+            ],
+          ),
+          Padding(
+            padding: const EdgeInsets.only(left: 30),
```
Comment (AbdalazizAbdallah) [inline lib/presentation/pages/drawer.dart:63]: 30.r in SizesApp

File: lib/presentation/pages/drawer.dart [hunk 10]
```diff
@@ -0,0 +1,133 @@
...
+            child: Column(
+              children: [
+                custom_ListTile(
+                    text: 'My_Order',
```
Comment (AbdalazizAbdallah) [inline lib/presentation/pages/drawer.dart:67]: try to make keys small char like my_order

**Review by AbdalazizAbdallah — CHANGES_REQUESTED** [review]
in the beginning Thank Abdalaziz for this

1- back icon you can use Icon class and select ios back, don't export anything you can do it
also a number of notifications

solve the conflict in code by pull statement to main branch the locally and again push here 
put your drawer in UI folder inside the widgets folder and use home page to test , after complete UI well separate folder according clean architecture , your work is very good but not at this time

ignore pubspec.lock by put *.lock in .gitignore file in any line

File: lib/presentation/controllers/drawer_controller.dart [hunk 11]
```diff
@@ -1,3 +1,4 @@
 import 'package:get/get.dart';
-
+import 'package:flutter/material.dart';
+final GlobalKey<ScaffoldState> drawerkey = GlobalKey();
```
Comment (AbdalazizAbdallah) [inline lib/presentation/controllers/drawer_controller.dart:3]: inside controller and access by controller declared in page ...
but in this case, I agree,

**Comment by AbdalazizAbdallah** [comment]
**in the beginning Thank Abdalaziz for this**

1- back icon you can use Icon class and select ios back, don't export anything you can do it
also a number of notifications

2- solve the conflict in code by pull statement to the main branch locally and again pushing here
3- put your drawer in the UI folder inside the widgets folder and use the home page to test, after complete UI well separate the folder according to to clean architecture, your work is very good but not at this time**###  till the rest of the members develop without conflict** 

4- ignore pubspec.lock by put *.lock in .gitignore file in any line

File: lib/presentation/controllers/drawer_controller.dart [hunk 12]
```diff
@@ -0,0 +1,4 @@
+import 'package:get/get.dart';
+import 'package:flutter/material.dart';
+final GlobalKey<ScaffoldState> drawerkey = GlobalKey();
```
Comment (fahjan) [inline lib/presentation/controllers/drawer_controller.dart:3]: move this to controller body

File: lib/presentation/pages/drawer.dart [hunk 13]
```diff
@@ -0,0 +1,133 @@
...
+                  color: ColorsApp.grey,
+                ),
+              ),
+            ],
+          ),
+          Padding(
+            padding:  EdgeInsets.only(left:SizesApp.r30),
+            child: Column(
+              children: [
+                custom_ListTile(
```
Comment (fahjan) [inline lib/presentation/pages/drawer.dart:66]: don't use underscore in custom widgets.

File: lib/presentation/widget/custom_ListTile.dart [hunk 14]
```diff
@@ -0,0 +1,34 @@
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:get/get.dart';
+import 'package:google_fonts/google_fonts.dart';
+
+import 'package:gtc_customer/core/constants/colors_app.dart';
+import 'package:gtc_customer/core/constants/sizes_app.dart';
+
+class custom_ListTile extends StatelessWidget {
```
Comment (fahjan) [inline lib/presentation/widget/custom_ListTile.dart:10]: use this please, CustomListTile

File: pubspec.lock [hunk 15]
```diff
@@ -0,0 +1,364 @@
+# Generated by pub
```
Comment (fahjan) [inline pubspec.lock:1]: ignore this file please.

**Review by mohammedalian — COMMENTED** [review]
resolve conflicts

**Pull request merged**